runtime/src/main/java/com/tns/AppConfig.java

```diff
@@ -98,23 +75,27 @@ public AppConfig(File appDir) {
                         if (dir.exists() && dir.isDirectory()) {
                             // this path is expected to be a directory, containing three sub-directories: armeabi-v7a, x86 and arm64-v8a
                             path = path + "/" + Build.CPU_ABI + "/" + KnownKeys.SnapshotFile.getName();
-                            values[KnownKeys.SnapshotFile.getIndex()] = path;
+                            values[KnownKeys.SnapshotFile.ordinal()] = path;
                         }
                     }
                     if (androidObject.has(KnownKeys.ProfilerOutputDirKey.getName())) {
-                        values[KnownKeys.ProfilerOutputDirKey.getIndex()] = androidObject.getString(KnownKeys.ProfilerOutputDirKey.getName());
+                        values[KnownKeys.ProfilerOutputDirKey.ordinal()] = androidObject.getString(KnownKeys.ProfilerOutputDirKey.getName());
                     }
                     if (androidObject.has(KnownKeys.GcThrottleTime.getName())) {
-                        values[KnownKeys.GcThrottleTime.getIndex()] = androidObject.getInt(KnownKeys.GcThrottleTime.getName());
+                        values[KnownKeys.GcThrottleTime.ordinal()] = androidObject.getInt(KnownKeys.GcThrottleTime.getName());
                     }
                     if (androidObject.has(KnownKeys.MemoryCheckInterval.getName())) {
-                        values[KnownKeys.MemoryCheckInterval.getIndex()] = androidObject.getInt(KnownKeys.MemoryCheckInterval.getName());
+                        values[KnownKeys.MemoryCheckInterval.ordinal()] = androidObject.getInt(KnownKeys.MemoryCheckInterval.getName());
                     }
                     if (androidObject.has(KnownKeys.FreeMemoryRatio.getName())) {
-                        values[KnownKeys.FreeMemoryRatio.getIndex()] = androidObject.getDouble(KnownKeys.FreeMemoryRatio.getName());
+                        values[KnownKeys.FreeMemoryRatio.ordinal()] = androidObject.getDouble(KnownKeys.FreeMemoryRatio.getName());
                     }
                     if (androidObject.has(KnownKeys.MarkingMode.getName())) {
-                        values[KnownKeys.MarkingMode.getIndex()] = androidObject.getString(KnownKeys.MarkingMode.getName());
+                        try {
+                            String value = androidObject.getString(KnownKeys.MarkingMode.getName());
+                            values[KnownKeys.MarkingMode.ordinal()] = MarkingMode.valueOf(value);
+                        } catch(Throwable e) {
```

Comment:
* catch Exception instead of throwable here if you don't want to have a specific warning for each exception
* don't leave catch block empty, please leave a warning that the default value will be used because we couldn't get the value for some reason

Reply:
Printed the stack trace and then a message with Log.v("JS", ... that the default will be used.